applications/serial_lte_modem/src/slm_at_tcp_proxy.c

```diff
@@ -492,14 +500,33 @@ int tcpsvr_input(int infd)
 			close(ret);
 			return -ECONNREFUSED;
 		}
-		/* Accept incoming connection */
 		LOG_DBG("accept(): %d", ret);
+
+		/* Whitelist filtering */
 		if (inet_ntop(AF_INET, &remote.sin_addr, peer_addr,
-			INET_ADDRSTRLEN) != NULL) {
-			sprintf(rsp_buf, "#XTCPSVR: %s connected\r\n",
-				peer_addr);
-			rsp_send(rsp_buf, strlen(rsp_buf));
+			INET_ADDRSTRLEN) == NULL) {
+			LOG_ERR("inet_ntop() failed: %d", -errno);
+			close(ret);
+			return -errno;
+		}
+		if (!proxy.whitelist) {
+			goto accept_done;
```

Comment:
I'm not in favour of multiplying `goto`s to control the flow of the application, it obfuscates the logic. 

How about adding a helper function, `static bool whitelist_allow(const char *peer_addr);` which would verify if whitelist is enabled and if so whether the address is on the list. It could return `true` if whitelist is disabled (hence all are allowed) or an IP address is on the list. `false` otherwise (IP blocked).

Then the whole logic with goto could be replaced with a simple
```
if (!whitelist_allow(peer_addr)) {
	LOG_WRN("Connection filtered");
	close(ret);
	return -ECONNREFUSED;
}
```

Reply:
Removed two goto in this function.